azure/src/test/java/com/azure/management/TestKubernetesCluster.java

```diff
@@ -27,8 +35,21 @@ public KubernetesCluster createResource(KubernetesClusters kubernetesClusters) t
         final String newName = "aks" + kubernetesClusters.manager().getSdkContext().randomResourceName("", 8);
         final String dnsPrefix = "dns" + newName;
         final String agentPoolName = "ap" + newName;
-        final String clientId = "clientId";
-        final String secret = "secret";
+        String clientId = "clientId";
+        String secret = "secret";
+
+        // aks can use another azure auth rather than original client auth to access azure service.
+        // Thus, set it to AZURE_AUTH_LOCATION_2 when you want.
+        String envSecondaryServicePrincipal = System.getenv("AZURE_AUTH_LOCATION_2");
+        if (envSecondaryServicePrincipal == null || envSecondaryServicePrincipal.isEmpty() || !(new File(envSecondaryServicePrincipal).exists())) {
+            envSecondaryServicePrincipal = System.getenv("AZURE_AUTH_LOCATION");
+        }
+
+        try {
+            HashMap<String, String> credentialsMap = ParseAuthFile(envSecondaryServicePrincipal);
+            clientId = credentialsMap.get("clientId");
+            secret = credentialsMap.get("clientSecret");
```

Comment:
Please double check the recorded json for id and secret.

Reply:
Yes, I've check for it. The clientId will be contained in record, and the secret will not. Secret is in the PUT body, and the record didn't record the request body.

Reply:
Like previous in the AKS test.